File: app/frontend/src/app/game/components/2_navigasi_menu/2_navigasi_bawah/5_geopolitik/2_organisasi_internasional/2_organisasi_regional/logic/python/membership_movement/1_join_logic/regional_join_handler.py
import random

class RegionalJoinHandler:
    """
    Handles the logic for an AI country to join a regional organization.
    Higher base probability than global orgs (50% vs 40%).
    """
    def __init__(self, base_probability=0.50):
        self.probability = base_probability

    def evaluate_join(self, country_name, decision_weight):
        """
        Returns True if the country decides to join.
        Regional alignment bonus is already included in decision_weight.
        """
        if decision_weight > 55:
            bonus = (decision_weight - 55) / 100
            adjusted_prob = min(self.probability + bonus, 0.90)
            
            if random.random() < adjusted_prob:
                return True
        return False

File: app/frontend/src/app/game/components/2_navigasi_menu/2_navigasi_bawah/5_geopolitik/2_organisasi_internasional/2_organisasi_regional/logic/python/membership_movement/1_join_logic/RegionalJoinHandler.ts
/**
 * RegionalJoinHandler TS Bridge
 */
export class RegionalJoinHandler {
    private probability: number;

    constructor(baseProbability = 0.50) {
        this.probability = baseProbability;
    }

    public evaluateJoin(countryName: string, decisionWeight: number): boolean {
        if (decisionWeight > 55) {
            const bonus = (decisionWeight - 55) / 100;
            const adjustedProb = Math.min(this.probability + bonus, 0.90);
            return Math.random() < adjustedProb;
        }
        return false;
    }
}
